docs/CONTRIBUTING.md

```diff
@@ -1,11 +1,327 @@
+# Contributing to OSTree
+
+The following guide is about OSTree forking, building, adding a command, testing the command, and submitting the change.
+
+- [Getting Started](#getting-started)
+- [Building OSTree](#building-ostree)
+    - [Install Build Dependencies](#install-build-dependencies)
+    - [OSTree Build Commands](#ostree-build-commands)
+- [Testing a Build](#testing-a-build)
+    - [Testing in a Container](#testing-in-a-container)
+    - [Testing in a Virtual Machine](#testing-in-a-virtual-machine)
+- [Tutorial: Adding a basic builtin command to OSTree](#tutorial-adding-a-basic-builtin-command-to-ostree)
+    - [Modifying OSTree](#modifying-ostree)
+    - [OSTree tests](#ostree-tests)
+    - [Submitting a patch](#submitting-a-patch)
+    - [Commit message style](#commit-message-style)
+    - [Returning workflow](#returning-workflow)
+
+---
+
+## [Getting Started](#getting-started)
+
+Fork https://github.com/ostreedev/ostree, the run the following commands.
+
+```bash
+$ git clone https://github.com/<username>/ostree
+$ git remote add upstream https://github.com/ostreedev/ostree
+$ git checkout master
+$ git branch --set-upstream-to=upstream/master master
+```
+Make a branch from master for your patch.
+```bash
+$ git checkout -b <name-of-branch>
+$ git branch --set-upstream-to=upstream/master <name-of-branch>
+```
+
+## [Building OSTree](#building-ostree)
+
+### [Install Build Dependencies](#install-build-dependencies)
+
+```bash
+$ sudo dnf builddep ostree
+$ sudo dnf install @buildsys-build
+```
+
+### [OSTree Build Commands](#ostree-build-commands)
+
+```bash
+git submodule update --init
+./autogen.sh --prefix=/usr --libdir=/usr/lib64 --sysconfdir=/etc
```

Comment:
Calling out the submodule init is fine, but note `autogen.sh` will also do it.  It also runs `./configure`, so you should omit that.

Reply:
see the new changes to this section.